lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -3664,10 +3654,10 @@ ConformanceChecker::inferTypeWitnessesViaValueWitness(ValueDecl *req,
   }
 
   auto setup = [&]() -> std::tuple<Optional<RequirementMatch>, Type, Type> {
-    fullWitnessType = removeSelfParam(witness, fullWitnessType);
+    fullWitnessType = fullWitnessType->removeSelfParam(witness);
     return std::make_tuple(
         None,
-        removeSelfParam(req, req->getInterfaceType()),
+        req->getInterfaceType()->removeSelfParam(req),
```

Comment:
There are other places we do stuff like getInterfaceType()->castTo<AnyFunctionType>()->getResult()... if you grep for variations on this you'll find other places to use the new utility method.

(Of course this is optional; I'm not asking you to do anything before merging the PR)